Core/Registry/IRegistryQuerier.cs

```diff
@@ -155,5 +155,54 @@ public static bool HasUpdate(this IRegistryQuerier querier, string identifier, K
             return !new List<string>(querier.InstalledDlls).Contains(identifier) && querier.IsInstalled(identifier, false)
                 && newest_version.version.IsGreaterThan(querier.InstalledVersion(identifier));
         }
+
+        /// <summary>
+        /// Is the mod installed and does it have a replaced_by relationship with a compatible version
+        /// Check latest information on installed version of mod "identifier" and if it has a "replaced_by"
+        /// value, check if there is a compatible version of the linked mod
+        /// </summary>
+        public static bool HasReplacement(this IRegistryQuerier querier, string identifier, KspVersionCriteria version)
+        {
+            // We only care about the installed version
+            CkanModule installedVersion;
+            try
+            {
+                installedVersion = querier.GetInstalledVersion(identifier);
+            }
+            catch (ModuleNotFoundKraken)
+            {
+                return false;
+            }
+            if (installedVersion == null) return false;  // Mod is not installed, so we don't care about replacements
+            //get the identifier from the replaced_by relationship, if it exists
+            RelationshipDescriptor replacedBy;
+            replacedBy = installedVersion.replaced_by;
+            if (replacedBy == null) return false; // No replaced_by relationship
+            // Now we need to see if there is a compatible version of the replacement
+            CkanModule replaceWith;
+            try
+            {
+                replaceWith = querier.LatestAvailable(replacedBy.name, version);
+            }
+            catch (ModuleNotFoundKraken)
+            {
+                return false;
+            }
+            if (replaceWith == null) return false;
+            // We definitely have a replacement mod which is compatible with our KSP Version Criteria
+            // If we have any mod version constrainsts, check them now.
+            if (replacedBy.min_version != null)
+            {
+                return !new List<string>(querier.InstalledDlls).Contains(identifier) && !replaceWith.version.IsLessThan(replacedBy.min_version);
+            }
+            else if (replacedBy.version != null)
+            {
+                return !new List<string>(querier.InstalledDlls).Contains(identifier) && replaceWith.version.Equals(replacedBy.version);
```

Comment:
I think this check can fail unnecessarily.

Suppose my currently installed mod has:
```
    "replaced_by": {
        "name": "MyMod",
        "version": "1.0.0"
    }
```
... but since that was set up, MyMod 1.1.1 has been released and indexed. The `LatestAvailable` call will find 1.1.1, and then reject it on this line because 1.0.0 and 1.1.1 are different versions. But 1.0.0 is still in the index and could be installed! I think we need a different call than `LatestAvailable` to find the CkanModule if `version` is set.

Reply:
Good point. Alternatively, we could make the spec only include min_version, which probably makes more sense.